### ISSUE TITLE
Class Magento\Sales\Model\Order\Customer\Management does not exist

### ISSUE THREAD

**Issue by andrew-smart** [magento/magento2#1693]
When trying to create an account from the checkout/onepage/success results in a 500.

Here is the stack trace;

```
#0 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Code/Reader/ClassReader.php(19): ReflectionClass->__construct('Magento\\Sales\\M...')
#1 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/Definition/Runtime.php(44): Magento\Framework\Code\Reader\ClassReader->getConstructor('Magento\\Sales\\M...')
#2 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(71): Magento\Framework\ObjectManager\Definition\Runtime->getParameters('Magento\\Sales\\M...')
#3 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Sales\\M...')
#4 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php(236): Magento\Framework\ObjectManager\ObjectManager->get('Magento\\Sales\\A...')
#5 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(53): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, 'Magento\\Sales\\A...', NULL, 'orderCustomerSe...', 'Magento\\Checkou...')
#6 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(82): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments('Magento\\Checkou...', Array, Array)
#7 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Checkou...', Array)
#8 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/App/ActionFactory.php(40): Magento\Framework\ObjectManager\ObjectManager->create('Magento\\Checkou...', Array)
#9 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/App/Router/Base.php(301): Magento\Framework\App\ActionFactory->create('Magento\\Checkou...', Array)
#10 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/App/Router/Base.php(161): Magento\Framework\App\Router\Base->matchAction(Object(Magento\Framework\App\Request\Http), Array)
#11 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/App/FrontController.php(41): Magento\Framework\App\Router\Base->match(Object(Magento\Framework\App\Request\Http))
#12 [internal function]: Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#13 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Interceptor.php(74): call_user_func_array(Array, Array)
#14 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#15 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#16 /Users/asmart/Sites/magebeta1/public_html/app/code/Magento/Store/App/FrontController/Plugin/RequestPreprocessor.php(89): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#17 [internal function]: Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#18 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(68): call_user_func_array(Array, Array)
#19 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#20 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#21 [internal function]: Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#22 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(68): call_user_func_array(Array, Array)
#23 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#24 /Users/asmart/Sites/magebeta1/public_html/app/code/Magento/PageCache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#25 [internal function]: Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#26 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Chain/Chain.php(68): call_user_func_array(Array, Array)
#27 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Interceptor.php(136): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#28 /Users/asmart/Sites/magebeta1/public_html/app/code/Magento/PageCache/Model/App/FrontController/BuiltinPlugin.php(73): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#29 [internal function]: Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#30 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/Interception/Interceptor.php(141): call_user_func_array(Array, Array)
#31 /Users/asmart/Sites/magebeta1/public_html/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#32 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/App/Http.php(115): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#33 /Users/asmart/Sites/magebeta1/public_html/lib/internal/Magento/Framework/App/Bootstrap.php(257): Magento\Framework\App\Http->launch()
#34 /Users/asmart/Sites/magebeta1/public_html/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#35 {main}
```

Doing a little digging there is a preference defined in the Magento/Sales/etc/di.xml;

```xml
<preference for="Magento\Sales\Api\OrderCustomerManagementInterface" type="Magento\Sales\Model\Order\Customer\Management" />
```

**Comment by sivaschenko** [comment]
Internal ticket MAGETWO-41816. Issue is already fixed.

**Comment by okorshenko** [comment]
@andrew-smart , Please, let us know if you still have an issue

**Comment by slavvka** [comment]
Hi @andrew-smart. Thank you for your report. 
The issue has been fixed in magento/magento2#29392 by @Nazar65 in 2.4-develop branch
Related commit(s):
 - [ebcdce406ee95ad7f6e9fe796846f99999aa2090](https://github.com/magento/magento2/commit/ebcdce406ee95ad7f6e9fe796846f99999aa2090)
 - [2534e462572b68d2f17317247c4caebed72de1bd](https://github.com/magento/magento2/commit/2534e462572b68d2f17317247c4caebed72de1bd)
 - [6c0934415fb249f60649d828089ba1140caba124](https://github.com/magento/magento2/commit/6c0934415fb249f60649d828089ba1140caba124)
 - [6344ad2a0d938c2a0dd964a21ca9a9222dd4a881](https://github.com/magento/magento2/commit/6344ad2a0d938c2a0dd964a21ca9a9222dd4a881)
 - [38562360f5a393ba84ed30c3a2e7140a3ab01e71](https://github.com/magento/magento2/commit/38562360f5a393ba84ed30c3a2e7140a3ab01e71)

The fix will be available with the upcoming 2.4.1 release.
